src/SamplesApp/UITests.Shared/Windows_Devices/MagnetometerTests.xaml.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Uno.UI.Samples.Controls;
+using Uno.UI.Samples.UITests.Helpers;
+using Windows.Devices.Sensors;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
+
+namespace UITests.Shared.Windows_Devices
+{
+	[SampleControlInfo("Windows.Devices", "Magnetometer", description: "Demonstrates use of Windows.Devices.Sensors.Magnetometer", viewModelType: typeof(MagnetometerTestsViewModel))]
+	public sealed partial class MagnetometerTests : UserControl
+	{
+		public MagnetometerTests()
+		{
+			this.InitializeComponent();
+		}
+	}
+
+	[Bindable]
+	public class MagnetometerTestsViewModel : ViewModelBase
+	{
+		private readonly Magnetometer _magnetometer = null;
+		private MagnetometerReading _lastReading;
+		private bool _readingChangedAttached;
+
+		public MagnetometerTestsViewModel(CoreDispatcher dispatcher) : base(dispatcher)
+		{
+			_magnetometer = Magnetometer.GetDefault();
+			if (_magnetometer != null)
+			{
+				_magnetometer.ReportInterval = 250;
+			}
+		}
+
+		public Command AttachReadingChangedCommand => new Command((p) =>
+		{
+			_magnetometer.ReadingChanged += Magnetometer_ReadingChanged;
```

Comment:
Even if it's only a sample app, this will leak. You should add `_magnetometer.ReadingChanged -= Magnetometer_ReadingChanged;` in the `ViewModelBase.Disposables` 

(in ctor: `Disposables.Add(Disposable.Create(() => _magnetometer.ReadingChanged -= Magnetometer_ReadingChanged));` )

Reply:
Done :-)